app/controllers/doctors_controller.rb

```diff
@@ -0,0 +1,41 @@
+class DoctorsController < ApplicationController
+  def index
+    doctors = Doctor.includes(:addresses).all
+    render json: doctors, include: :addresses
+  end
+
+  def new
+    Doctor.new
+  end
+
+  def create
+    doctor = Doctor.new(doctor_params)
+    if doctor.save
+      render json: doctor, include: :addresses, status: :created
+    else
+      render json: { error: doctor.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  def show
+    user = User.find(params[:user_id])
+    doctor = user.doctors.find(params[:id])
+    render json: doctor, status: :ok
+  end
+
+  def destroy
+    doctor = Doctor.find(params[:id])
+    doctor.destroy
+    head :no_content
+  end
```

Comment:
- [ ] What happens if the doctor is not successfully destroyed?  Kindly add some error handling here. 👍